admin/admin.go

```diff
@@ -145,6 +160,32 @@ func (m *RPCMethods) GetL1Addresses(args *uint64) *map[string]string {
 		"ProxyAdmin":                        chain.L2Config.L1Addresses.ProxyAdmin.String(),
 		"SuperchainConfig":                  chain.L2Config.L1Addresses.SuperchainConfig.String(),
 	}
-	m.Log.Debug("admin_getL2Addresses")
+	m.log.Debug("admin_getL2Addresses")
 	return &reply
 }
+
+func (m *RPCMethods) GetL2ToL2MessageByMsgHash(args *common.Hash) *interop.L2ToL2Message {
+
+	if args == nil {
+		m.log.Error("valid msg hash not provided", "args", *args)
+		return nil
+	}
+
+	if m.networkConfig.InteropEnabled && m.l2ToL2MsgIndexer != nil {
+
+		storeEntry, err := m.l2ToL2MsgIndexer.Get(*args)
+
+		if err != nil {
+			m.log.Error("failed to get msg", "error", err)
+			return nil
+		}
+
+		reply := storeEntry.Message()
+
+		return reply
+
+	} else {
+		m.log.Error("auto relay not enabled", "args", *args)
+		return nil
+	}
```

Comment:
the indexer runs regardless of the auto relayer. maybe some thing like "Interop is not enabled, please enable interop with --interop.enabled"

Reply:
also this should not be a log it should return an RPC error with the correct code